src/main/java/com/manywho/services/saml/services/AuthenticationService.java

```diff
@@ -25,20 +28,30 @@ public AuthenticatedWhoResult createAuthenticatedWhoResultWithError(String error
     }
 
     public AuthenticatedWhoResult createAuthenticatedWhoResult(ApplicationConfiguration configuration, SamlResponseHandler response) throws Exception {
+        return createAuthenticatedWhoResult(configuration, response, true);
+    }
+
+    public AuthenticatedWhoResult createAuthenticatedWhoResult(ApplicationConfiguration configuration, SamlResponseHandler response, boolean validate) throws Exception {
```

Comment:
I understand that the validate parameter is used only because it is needed in the tests, I think that could be removed and simplified, but you will need to split this test in two tests:

First testing the method createAuthenticatedWhoResult using a mock of the handler. You can change dates as you wish with that test and also test the errors.

The logic of the other test is already covered in the current test, it would be to get from the SAML Response the sessionNotAfter.
e.g.
SamlResponseHandler handler = samlService.decryptResponse(configuration, samlResponse, "https://flow.manywho.com/api/run/1/saml");

assert that handler.getResponse().getSessionNotAfter()  != null